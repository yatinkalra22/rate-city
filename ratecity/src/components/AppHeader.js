import React, { Component } from "react";
import "../styles/AppHeader.css";

export default class AppHeader extends Component {
  render() {
    return (
      <div className="App-header-container">
        <div className="headline-text" id="top-home-loan-testcase">
          Top Home Loan Products
        </div>
        <div className="headline-menu">
          <div id="all-testcase">ALL</div>
          <div id="refinance-testcase">REFINANCE</div>
          <div id="fixed-rate-testcase">FIXED RATE</div>
          <div id="first-home-buyer-testcase">FIRST HOME BUYER</div>
          <div id="nvestor-testcase">NVESTOR</div>
          <div id="next-home-buyer-testcase">NEXT HOME BUYER</div>
        </div>
      </div>
    );
  }
}
