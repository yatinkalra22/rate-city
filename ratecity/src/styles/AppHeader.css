.App-header-container {
  border-bottom: 1px solid lightblue;
}
.App-header-container .headline-text {
  font-size: 15px;
  text-align: center;
  font-weight: 600;
  padding: 10px;
}
.App-header-container .headline-menu {
  display: flex;
  font-size: 10px;
  justify-content: space-between;
  color: #333333;
}

.App-header-container .selectedMenu {
  color: var(--secondary-color);
  font-weight: 600;
}
.App-header-container .unselectedMenu {
  color: #473d3d;
}

@media (min-width: 580px) {
  .App-header-container {
    display: flex;
    justify-content: space-between;
  }
  .App-header-container .headline-menu {
    font-size: 12px;
    padding: 10px;
  }
  .App-header-container .headline-menu > div {
    padding: 0 5px;
  }
}
