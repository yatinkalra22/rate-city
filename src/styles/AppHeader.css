.App-header-container {
  border-bottom: 1px solid lightblue;
  height: 20%;
}
.App-header-container .headline-text {
  font-size: 18px;
  text-align: center;
  font-weight: 600;
  padding: 10px;
}
.App-header-container .headline-menu {
  display: flex;
  font-size: 10px;
  justify-content: space-between;
  color: #333333;
  font-size: 14px;
    padding: 10px;
}

.App-header-container .selectedMenu {
  color: var(--secondary-color);
  font-weight: 600;
}
.App-header-container .unselectedMenu {
  color: #473d3d;
}

@media (min-width: 580px) {
  .App-header-container {
    display: flex;
    justify-content: space-between;
  }
  .App-header-container .headline-menu {
    font-size: 15px;
    
  }
  .App-header-container .headline-menu > div {
    padding: 0 5px;
  }
}
