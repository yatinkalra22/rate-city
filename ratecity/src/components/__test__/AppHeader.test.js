import React from "react";
import { shallow } from "enzyme";
import AppHeader from "../../components/AppHeader";

const wrapper = shallow(<AppHeader />);

describe("AppHeader Component", () => {
  it("should render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should have 'Top Home Loan Products' as the headerText ", () => {
    expect(wrapper.find("#top-home-loan-testcase").text()).toBe(
      "Top Home Loan Products"
    );
  });
  it("should have multiple Text in the headLine menu ", () => {
    expect(wrapper.find("#all-testcase").text()).toBe("ALL");
    expect(wrapper.find("#refinance-testcase").text()).toBe("REFINANCE");
    expect(wrapper.find("#fixed-rate-testcase").text()).toBe("FIXED RATE");
    expect(wrapper.find("#first-home-buyer-testcase").text()).toBe(
      "FIRST HOME BUYER"
    );
    expect(wrapper.find("#nvestor-testcase").text()).toBe("NVESTOR");
    expect(wrapper.find("#next-home-buyer-testcase").text()).toBe(
      "NEXT HOME BUYER"
    );
  });
});
