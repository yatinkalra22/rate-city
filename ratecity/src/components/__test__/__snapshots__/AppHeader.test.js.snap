// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader Component should render the component 1`] = `ShallowWrapper {}`;
